src/main/kotlin/com/blincke/commune_api/repositories/UserRepository.kt

```diff
@@ -15,37 +15,71 @@ interface UserRepository : JpaRepository<User, String> {
+                "and u != (?1) " +
+                "and lower(u.name) like %?2%"
     )
-    fun selectSuggestedFriends(user: User, pageable: Pageable): List<User>
+    fun selectSuggestedFriends(
+        user: User,
+        nameContainsLower: String,
+        pageable: Pageable,
+    ): List<User>
 
+    /**
+     * select f1.recipient
+     * from friendship f1
+     * inner join friendship f2
+     * on f1.requester = f2.recipient
+     * and f1.recipient = f2.requester
+     * and f1.requester = 'userId';
+     */
     @Query(
-        value = "select u from User u " +
-                "where u in " +
-                "(select f.friendshipId.requester from Friendship f " +
-                "where f.friendshipId.recipient = (?1) and f.status = 'ACCEPTED')" +
-                "or u in " +
-                "(select f.friendshipId.recipient from Friendship f " +
-                "where f.friendshipId.requester = (?1) and f.status = 'ACCEPTED')" +
-                "and u != (?1)"
+        value = "select f1.friendshipId.recipient " +
+                "from Friendship f1 " +
+                "inner join Friendship f2 " +
+                "on f1.friendshipId.recipient = f2.friendshipId.requester " +
+                "and f2.friendshipId.recipient = f1.friendshipId.requester " +
+                "and f1.friendshipId.requester = (?1)"
     )
     fun selectConfirmedFriends(user: User, pageable: Pageable): List<User>
 
+    /**
+     *  select f1.recipient
+     *  from friendship f1
+     *  where f1.requester = 'a'
+     *  and 'a' not in
+     *  (select f2.recipient
+     *  from friendship f2
+     *  where f2.requester=f1.recipient);
+     */
     @Query(
-        value = "select u from User u " +
-                "where u in " +
-                "(select f.friendshipId.recipient from Friendship f " +
-                "where f.friendshipId.requester = (?1) and f.status = 'PENDING')" +
-                "and u != (?1)"
+        value = "select f1.friendshipId.recipient " +
+                "from Friendship f1 " +
+                "where f1.friendshipId.requester = (?1) " +
+                "and (?1) not in " +
+                "(select f2.friendshipId.recipient " +
+                "from Friendship f2 " +
+                "where f2.friendshipId.requester = f1.friendshipId.recipient)"
     )
     fun selectFriendshipsThatArePendingThatISent(user: User, pageable: Pageable): List<User>
 
+
+    /**
+     *  select f1.requester
+     *  from friendship f1
+     *  where f1.recipient = 'a'
+     *  and 'a' not in
+     *  (select f2.requester
+     *  from friendship f2
+     *  where f2.recipient=f1.requester);
+     */
     @Query(
-        value = "select u from User u " +
-                "where u in " +
-                "(select f.friendshipId.requester from Friendship f " +
-                "where f.friendshipId.recipient = (?1) and f.status = 'PENDING')" +
-                "and u != (?1)"
+        value = "select f1.friendshipId.requester " +
```

Comment:
Finds all occurences of friendships where (recipient, requester) exists but (requester, recipient) does not exist and requester = userId.